test/addons/openssl-client-cert-engine/testengine.cc

```diff
@@ -0,0 +1,111 @@
+#include <assert.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include <openssl/engine.h>
+#include <openssl/pem.h>
+
+#include <string>
+
+#ifndef ENGINE_CMD_BASE
+# error did not get engine.h
+#endif
+
+#define TEST_ENGINE_ID      "testengine"
+#define TEST_ENGINE_NAME    "dummy test engine"
+
+#define AGENT_KEY           "test/fixtures/keys/agent1-key.pem"
+#define AGENT_CERT          "test/fixtures/keys/agent1-cert.pem"
+
+static int EngineInit(ENGINE* engine) {
+  return 1;
+}
+
+static int EngineFinish(ENGINE* engine) {
+  return 1;
+}
+
+static int EngineDestroy(ENGINE* engine) {
+  return 1;
+}
+
+static std::string LoadFile(const char* filename) {
+  std::string ret;
+  FILE* fp = fopen(filename, "r");
+  if (fp == NULL) {
+    printf("Could not open file '%s'\n", filename);
+    return ret;
+  }
+
+  char buf[1024];
+  while (true) {
+    size_t r = fread(buf, 1, 1024, fp);
+    if (r > 0) {
+       ret.append(buf, r);
+    }
+    if (r < 1024) {
+      break;
+    }
+  }
+
+  fclose(fp);
+  return ret;
+}
```

Comment:
This could be shortened somewhat to this:
```c++
#include <fstream>
#include <iterator>
#include <string>
// ...
static std::string LoadFile(const char* filename) {
  std::ifstream file(filename);
  return std::string(std::istreambuf_iterator<char>(file),
                     std::istreambuf_iterator<char>());
}
```

Reply:
As well, if you wrap everything in a `namespace { }  // namespace anonymous`, you can drop the `static` keywords.